pkg/tests/cross-tests/tfwrite.go

```diff
@@ -38,6 +39,87 @@ func WriteHCL(out io.Writer, sch map[string]*schema.Schema, resourceType, resour
 	return err
 }
 
+type PFWriter struct {
+	out io.Writer
+}
+
+func WritePF(out io.Writer) PFWriter { return PFWriter{out} }
+
+func (w PFWriter) Provider(sch pfproviderschema.Schema, providerName string, config map[string]cty.Value) error {
+	if !cty.ObjectVal(config).IsWhollyKnown() {
+		return fmt.Errorf("WriteHCL cannot yet write unknowns")
+	}
+	f := hclwrite.NewEmptyFile()
+	block := f.Body().AppendNewBlock("provider", []string{providerName})
+	writePfProvider(block.Body(), sch, config)
+	_, err := f.WriteTo(w.out)
+	return err
+}
+
+func writePfProvider(body *hclwrite.Body, schemas pfproviderschema.Schema, values map[string]cty.Value) {
+	writePfObject(body, pfproviderschema.NestedBlockObject{
+		Attributes: schemas.Attributes,
+		Blocks:     schemas.Blocks,
+	}, values)
+}
+
+func writePfBlock(key string, body *hclwrite.Body, schemas pfproviderschema.Block, value cty.Value) {
+	switch schemas := schemas.(type) {
+	case *pfproviderschema.ListNestedBlock:
+		for _, v := range value.AsValueSlice() {
+			b := body.AppendNewBlock(key, nil).Body()
+			writePfObject(b, schemas.NestedObject, v.AsValueMap())
+		}
+	case *pfproviderschema.SetNestedBlock:
+		values := value.AsValueSet().Values()
+		for _, v := range values {
+			b := body.AppendNewBlock(key, nil).Body()
+			writePfObject(b, schemas.NestedObject, v.AsValueMap())
+		}
+	case *pfproviderschema.SingleNestedBlock:
+		body := body.AppendNewBlock(key, nil).Body() // This goes in directly, so just overwrite body
```

Comment:
I don't understand this comment, can you clarify?

Reply:
Clarified in a comment on `writePfBlock`.